src/checkin/services/flow/step/GetExtraPassengerInformationStep.ts

```diff
@@ -0,0 +1,42 @@
+import { Builder } from 'builder-pattern'
+import { Context } from '../../../model/Context'
+import { Session } from '../../../model/session'
+import { UserInformation } from '../../../model/session/UserInformation'
+import StepTemplate from '../StepTemplate'
+import { SessionData } from '../../../model/session/SessionData'
+
+export default class GetExtraPassengerInformationStep extends StepTemplate {
+
+  when(context: Context): boolean {
+    const numberOfPassengers = context.getSession().data.passengers
+    return numberOfPassengers > 1
+  }
+
+  onExecute(context: Context): Promise<boolean> {
```

Comment:
The onExecute method appears to handle multiple responsibilities, such as accessing session and request data, performing validation, and updating session information. Breaking down this method into smaller, more focused methods would improve readability and maintainability.

Reply:
For the accessing session and request data I could encapsulate both in its respective methods but I think it would be kinda redundant because of the `getSession()` and `getRequest() `methods already describes what returns.

However it's true that for the validation and session handling, its readability can be improved by breaking them down on its respective methods. Thanks!